apollo-router-core/src/services/mod.rs

```diff
@@ -92,9 +92,19 @@ pub struct SubgraphRequest {
     pub http_request: http_compat::Request<Request>,
 
     pub context: Context,
+
+    pub operation_kind: OperationKind,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum OperationKind {
```

Comment:
this will be replaced by rebasing over the changes of @o0Ignition0o

Reply:
ah, does this render my earlier comment about having an enum for operation_kind redundant? Why don't we use this instead of the String up above?

Reply:
ah, it's missing the small serde layer to deserialize it directly to the enum, I'll add it

Reply:
done in 941c2aad8f8bc384b77e14704712799269c4642f